core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -370,6 +390,13 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
             }
 
             @Override
+            void validateParsed(Number value) {
+                if (!Double.isFinite(value.doubleValue())) {
+                    throw new IllegalArgumentException("[double] supports only finite values, but got [" + value.toString() + "]");
```

Comment:
Can we use this message format where we show what we got as an invalid value in the error messages for the other types too?